tokenizers/src/models/unigram/model.rs

```diff
@@ -401,16 +406,33 @@ impl Model for Unigram {
     fn get_vocab_size(&self) -> usize {
         self.vocab.len()
     }
-
     fn tokenize(&self, sentence: &str) -> Result<Vec<Token>> {
         let str_tokens = self.encode(sentence)?;
         let mut offset = 0;
         let mut tokens = Vec::with_capacity(str_tokens.len());
         for string in str_tokens {
             let id: u32 = match self.token_to_ids.get(&string) {
                 Some(id) => *id,
-                None => self.unk_id.ok_or(UnigramError::MissingUnkId)? as u32,
+                None => {
+                    if self.byte_fallback {
+                        let mut ids = Vec::new();
+                        for byte in string.as_bytes() {
+                            let code = format!("<{:#04X}>", byte);
+                            if let Some(id) = self.token_to_ids.get(&code) {
+                                ids.push(*id);
+                            }
+                        }
+                        if ids.is_empty() {
+                            self.unk_id.ok_or(UnigramError::MissingUnkId)? as u32
+                        } else {
+                            *ids.first().unwrap()
+                        }
```

Comment:
probably wrong, all tokens have to be pushed + offsets

Reply:
atleast in sentencepiece, the bytefallbacks do not use the push/offset trick and just use reserved
<0x00> to <0xFF>